h/api/schemas.py

```diff
@@ -264,17 +192,50 @@ class CreateAnnotationSchema(object):
     """Validate the POSTed data of a create annotation request."""
 
     def __init__(self, request):
-        self.structure = AnnotationSchema(request)
+        self.structure = AnnotationSchema()
+        self.request = request
 
     def validate(self, data):
-        return self.structure.validate(data)
+        appstruct = self.structure.validate(data)
+
+        new_appstruct = {}
+
+        _remove_protected_fields(appstruct)
+
+        new_appstruct['userid'] = self.request.authenticated_userid
+        new_appstruct['target_uri'] = appstruct.pop('uri', u'')
```

Comment:
If postgres off, the legacy storage will let me create an annotation with no uri _but_ it puts `null` rather than `''` as the default :(

```json
~ http POST 'http://localhost:5000/api/annotations' Authorization:'Bearer 6879-e967d22983d7f00b91fadf1a62c2e52c' text=hi
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Connection: close
Content-Length: 564
Content-Type: application/json; charset=UTF-8
Date: Fri, 06 May 2016 10:20:51 GMT
Server: gunicorn/19.4.5

{
    "created": "2016-05-06T10:20:51.159868+00:00",
    "document": {},
    "group": "__world__",
    "id": "AVSFlYxYsSgLSOURFTGL",
    "links": {
        "html": "http://localhost:5000/a/AVSFlYxYsSgLSOURFTGL",
        "json": "http://localhost:5000/api/annotations/AVSFlYxYsSgLSOURFTGL"
    },
    "nipsa": true,
    "permissions": {
        "admin": [
            "acct:fred@localhost"
        ],
        "delete": [
            "acct:fred@localhost"
        ],
        "read": [
            "group:__world__"
        ],
        "update": [
            "acct:fred@localhost"
        ]
    },
    "tags": [],
    "target": [
        {
            "source": null
        }
    ],
    "text": "hi",
    "updated": "2016-05-06T10:20:51.159883+00:00",
    "uri": null,
    "user": "acct:fred@localhost"
}

~ 
```